packages/grafana-ui/src/components/Segment/SegmentInput.story.tsx

```diff
@@ -6,20 +6,20 @@ import { SegmentInput, Icon, SegmentSection } from '@grafana/ui';
 
 import { SegmentInputProps } from './SegmentInput';
 
-const SegmentFrame = ({ children }: any) => (
+const SegmentFrame = ({ children }: React.PropsWithChildren) => (
   <>
     <SegmentSection label="Segment Name">{children}</SegmentSection>
   </>
 );
 
 export const BasicInput = () => {
-  const [value, setValue] = useState('some text');
+  const [value, setValue] = useState<string | number>('some text');
```

Comment:
When can this be a number?

Reply:
`SegmentInput`s `onChange` function is called with an argument` that is `string | number`. we _could_ coerce this to a string when `setValue` is called, e.g. `setValue(String(text))` and leave the state variable as a string, but 🤷